### PR TITLE
frontegg-client: client is now able to return claims + role id type

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

This PR enables the `frontegg-client` to decode and return the user `claims` (used by CLI to retrieve an app-password email) using the well-known endpoint. Additionally, turns the roles id type to Uuid, to decode/handle roles more simply.  

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/frontegg-client/src/client.rs [hunk 1]
```diff
@@ -205,4 +207,29 @@ impl Client {
...
+        let req = self.build_request(Method::GET, well_known);
+        let jwks: JwkSet = self.send_request(req).await?;
+        Ok(jwks)
+    }
+
+    /// Verifies the JWT signature using a JWK from the well-known endpoint and
+    /// returns the user claims.
+    pub async fn claims(&self) -> Result<Claims, Error> {
+        let jwks = self.get_jwks().await?;
+        let jwk = jwks.keys.first().expect("Error validating signature JWK.");
```
Comment (benesch) [inline src/frontegg-client/src/client.rs:223]:
```suggestion
        let jwk = jwks.keys.first().ok_or_else(|| anyhow!("JWK set contained no keys"));
```

File: src/frontegg-client/src/client.rs [hunk 2]
```diff
@@ -205,4 +207,29 @@ impl Client {
...
+        let well_known = vec![".well-known", "jwks.json"];
+        let req = self.build_request(Method::GET, well_known);
+        let jwks: JwkSet = self.send_request(req).await?;
+        Ok(jwks)
+    }
+
+    /// Verifies the JWT signature using a JWK from the well-known endpoint and
+    /// returns the user claims.
+    pub async fn claims(&self) -> Result<Claims, Error> {
+        let jwks = self.get_jwks().await?;
```
Comment (benesch) [inline src/frontegg-client/src/client.rs:222]:
```suggestion
        let jwks = self.get_jwks().await.context("fetching JWK set")?;
```

File: src/frontegg-client/src/client.rs [hunk 3]
```diff
@@ -205,4 +207,29 @@ impl Client {
...
+        let jwk = jwks.keys.first().expect("Error validating signature JWK.");
+        let token = self.auth().await?;
+
+        let token_data = decode::<Claims>(
+            &token,
+            &DecodingKey::from_jwk(&jwk).unwrap(),
```
Comment (benesch) [inline src/frontegg-client/src/client.rs:228]:
```suggestion
            &DecodingKey::from_jwk(&jwk).context("converting JWK into decoding key")?,
```

File: src/frontegg-client/src/client.rs [hunk 4]
```diff
@@ -205,4 +207,29 @@ impl Client {
...
+            &Validation::new(Algorithm::RS256),
+        )
+        .unwrap();
```
Comment (benesch) [inline src/frontegg-client/src/client.rs:231]:
```suggestion
        .context("decoding JWT claims")?;
```

**Review by benesch — COMMENTED** [review]
Nice! Just a bunch of suggestions around gracefully handling errors rather than panicking. Because the JWK set is returned via a network API request, we have to assume that Frontegg returned malformed data (maybe a transient network failure), and so should be prepared to gracefully report any such errors that we discover.

**Comment by joacoc** [comment]
@benesch Done. Thank you! I forgot about those. I created a custom error for each.

**Review by benesch — APPROVED** [review]

**Pull request merged**